liota/dccs/iotcc.py

```diff
@@ -236,6 +242,12 @@ def _properties(self, msg_id, entity_type, entity_id, entity_name, timestamp, pr
             }
         }
         for key, value in properties.items():
+            if not self._validate_input(key):
+                log.error("Property key {0} contains unacceptable character for resource {1}".format(key, entity_name))
+                continue
```

Comment:
What is the reason for not raising ValueError here?

Reply:
Changed Exception to ValueError

Reply:
@GregBollella @KohliDev @pmasrani 
I have updated the PR with limit checks for the number of characters for resource names and properties.
These limits are from airwatch and relevant for v1.
Could you please confirm if this is acceptable.

Reply:
@arunv10 You are logging the error not raising it is this the expected behavior? I think if it is an important property which user want to set but due to the restriction, it fails we should raise the exception to user code.

Reply:
@KohliDev I think it would be better to do the best effort approach for setting properties as there are optional. So that other valid properties are set. What do you think?

Reply:
Okay got it. Then basically, we shouldn't set the property if they don't match the criteria and log the error. 

But the current code changes in this PR will still send the property, I think we shouldn't as we know the limit exist on the AW (v1). Also `continue` statement shouldn't be required for logging the error in `if` Loop.

Reply:
In the code changes, if a property's name or value doesn't satisfy the criteria then it is skipped and the next property in the for loop is picked using the continue statement.

If a property passes all tests then it is appended to the property data of the add_properties_request message.

Reply:
Good catch Vaibhav!

Arun, i think we should throw the error in this case as well, instead of best effort. If we just log the error, the property will nto get set, but the user call to set properties will succeed. He will have no way to know if the property got set or not, but to go through the logs. If we throw the error, the loading of the package which is setting the property will fail, and he will be forced to take a look at it.

Reply:
@KohliDev and @pmasrani  I understood your point. I have addressed them.